admin/server/service.go

```diff
@@ -98,18 +204,139 @@ func (s *Server) UpdateService(ctx context.Context, req *adminv1.UpdateServiceRe
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
-	updatedService, err := s.admin.DB.UpdateService(ctx, service.ID, &database.UpdateServiceOptions{
-		Name: valOrDefault(req.NewName, service.Name),
```

Comment:
Is there a reason it doesn't use `valOrDefault` for `NewName` and `Attributes` now?

Reply:
it was kind of defensive if the map parameter remains `nil` instead of empty map then we can use it. Will check

Reply:
Interesting thing is if I set `'{}'` in the cli as attributes, cmd initializes the map but in the admin service the request attribute is nil so hard to distinguish between empty and no response. I was thinking if user sets `'{}'` then we might delete the attribute. One way is to accept attrs as string in the rpc and deserialize there, other is to add another method to clear all attributes.